### PR TITLE
[boost-asio] Fix duplicate symbols of asio::dummy_return<void>

### PR DESCRIPTION
**Pull request by cedral**
**Describe the pull request**
The fix already exists (chriskohlhoff/asio#584), but it hasn't been merged yet. Would it be possible to apply a patch to fix this for version 1.75.0? It is very low impact https://gist.github.com/jprochazk/ecb261bb99489a3d0fef03a4ae27c064

- What does your PR fix? Fixes #
Build any of the asio coroutine examples with latest MSVC and Boost 1.75.0. It should compile, but linking will fail:
```error LNK2005: "void __cdecl boost::asio::dummy_return<void>(void)" (??$dummy_return@X@asio@boost@@YAXXZ) already defined```

- Which triplets are supported/not supported? Have you updated the CI baseline?
*-windows-* is affected, no I haven't.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
I haven't read it.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@cedral Thanks for this PR!
Can you sign CLA first?

Thanks.

File: ports/boost-asio/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Automatically generated by scripts/boost/generate-ports.ps1
 Source: boost-asio
-Version: 1.75.0
+Version: 1.75.0.1
```
Comment (JackBoosY) [inline ports/boost-asio/CONTROL:3]:
```suggestion
Version: 1.75.0
Port-Version: 1
```

File: ports/boost-asio/portfile.cmake [hunk 2]
```diff
@@ -6,7 +6,9 @@ vcpkg_from_github(
     REF boost-1.75.0
     SHA512 5f332644207f87af8ffbf93d9f1f72b7ca6dfa96d77a871308951c492ae27fac20bad2bfdf4e1de06585c1405d1875c4d4335786b3fc08aa0a76ef4de3626c27
     HEAD_REF master
-    PATCHES windows_alloca_header.patch
+    PATCHES
+        windows_alloca_header.patch
+	inline_dummy_return.patch
```
Comment (JackBoosY) [inline ports/boost-asio/portfile.cmake:11]:
```suggestion
        windows_alloca_header.patch
        inline_dummy_return.patch
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by cedral** [comment]
Thanks I'll take note of the Port-Version setting for future reference. I am not sure if I am supposed to close this or not so I will leave that to you.

**Comment by JackBoosY** [comment]
LGTM.

**Pull request merged**